server/routes/browser.py

```diff
@@ -29,7 +31,11 @@ def browser_main():
 
 @bp.route('/<path:dcid>')
 def browser_node(dcid):
-    node_name = shared_api.names([dcid]).get(dcid)
-    if not node_name:
-        node_name = dcid
+    node_name = dcid
+    try:
+        node_name = shared_api.names([dcid]).get(dcid)
+        if not node_name:
+            node_name = dcid
```

Comment:
since `node_name` is already `dcid` by default, maybe instead:

```
try:
  api_name = shared_api.names...
  if api_name:
    node_name = api_name
```

Reply:
Done